app/src/lib/git/log.ts

```diff
@@ -106,58 +104,26 @@ export async function getCommits(
     return new Array<Commit>()
   }
 
-  const out = result.stdout
-  const lines = out.split('\0')
-  // Remove the trailing empty line
-  lines.splice(-1, 1)
-
-  if (lines.length === 0) {
-    return []
-  }
-
   const trailerSeparators = await getTrailerSeparatorCharacters(repository)
+  const parsed = parse(result.stdout)
 
-  const commits = lines.map(line => {
-    const pieces = line.split(delimiterString)
-    const sha = pieces[0]
-    const shortSha = pieces[1]
-    const summary = pieces[2]
-    const body = pieces[3]
-    const authorIdentity = pieces[4]
-    const committerIdentity = pieces[5]
-    const shaList = pieces[6]
-
-    const parentSHAs = shaList.length ? shaList.split(' ') : []
-    const trailers = parseRawUnfoldedTrailers(pieces[7], trailerSeparators)
-    const tags = getCaptures(pieces[8], /tag: ([^\s,]+)/g)
+  return parsed.map(commit => {
+    const tags = getCaptures(commit.refs, /tag: ([^\s,]+)/g)
       .filter(i => i[0] !== undefined)
       .map(i => i[0])
-    const author = CommitIdentity.parseIdentity(authorIdentity)
-
-    if (!author) {
-      throw new Error(`Couldn't parse author identity for '${shortSha}'`)
-    }
-
-    const committer = CommitIdentity.parseIdentity(committerIdentity)
-
-    if (!committer) {
-      throw new Error(`Couldn't parse committer identity for '${shortSha}'`)
-    }
 
     return new Commit(
```

Comment:
This is beautiful! 😍

Reply:
... meant to grab a set of lines here -> the refactor of log is what I was referring too. 🤣